wait.go

```diff
@@ -108,23 +109,36 @@ var urlRegexp = regexp.MustCompile("^(https?://)?(?P<host>.+):(?P<port>[0-9]+)(.
 // once it is reachable.
 func WithWaitForTCP(addr string) WaitOption {
 	return func(cfg waitConfig) waitConfig {
-		cfg.waiters = append(cfg.waiters,
-			tcpWaiter{
-				addr:   urlRegexp.ReplaceAllString(addr, "$host:$port"),
-				logger: cfg.logger,
-			},
-		)
+		cfg.waiters = append(cfg.waiters, netWaiter{
+			addr:    urlRegexp.ReplaceAllString(addr, "$host:$port"),
+			logger:  cfg.logger,
+			network: tcp,
+		})
 		return cfg
 	}
 }
 
-type tcpWaiter struct {
-	addr   string
-	logger *slog.Logger
+// WithWaitForUnix makes a new unix waiter that will ping a socket and return
```

Comment:
Nit: This godoc doesn't document the format of the unix addr.

If we go with my other comment, I believe the format can match the standard library's format used by `net` to stay compatible.